### PR TITLE
configure_upgrade_step in e2e.sh, and use it for kubernetes-upgrade-gke

### PR DESCRIPTION
**Pull request by ikehz**
@kubernetes/goog-testing

### CODE REVIEW

**Comment by ihmccreery** [comment]
@ixdy This is ready for review.

**Comment by ihmccreery** [comment]
This is keeping https://github.com/kubernetes/kubernetes/issues/17734#issuecomment-159709543 in mind.

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -65,6 +65,161 @@ function get_latest_trusty_image() {
     rm -rf .gsutil &> /dev/null
 }
 
+# Properly configure globals for an upgrade step in a GKE upgrade suite
```
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: GKE/GCE, or is this just for GKE?

**Comment by jlowdermilk** [comment]
lgtm, needs rebase

**Comment by ihmccreery** [comment]
Rebased & squashed.

File: hack/jenkins/e2e.sh (hunk 1 above)
Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Done.

**Pull request merged**